internal/weldr/api.go

```diff
@@ -2165,13 +2165,66 @@ func (api *API) depsolveBlueprintForImageType(bp blueprint.Blueprint, options di
 }
 
 func (api *API) resolveContainersForImageType(bp blueprint.Blueprint, imageType distro.ImageType) ([]container.Spec, error) {
-	resolver := container.NewResolver(imageType.Arch().Name())
 
-	for _, c := range bp.Containers {
-		resolver.Add(c.Source, c.Name, c.TLSVerify)
+	specs := make([]container.Spec, len(bp.Containers))
+
+	// shortcut
+	if len(bp.Containers) == 0 {
+		return specs, nil
+	}
+
+	job := worker.ContainerResolveJob{
+		Arch:  api.archName,
+		Specs: make([]worker.ContainerSpec, len(bp.Containers)),
+	}
+
+	for i, c := range bp.Containers {
+		job.Specs[i] = worker.ContainerSpec{
+			Source:    c.Source,
+			Name:      c.Name,
+			TLSVerify: c.TLSVerify,
+		}
+	}
+
+	jobId, err := api.workers.EnqueueContainerResolveJob(&job, "")
+
+	if err != nil {
+		return specs, err
+	}
+
+	var result worker.ContainerResolveJobResult
+
+	for {
+		status, _, err := api.workers.ContainerResolveJobStatus(jobId, &result)
+
+		if err != nil {
+			return specs, err
+		}
+
+		if result.JobError != nil {
+			return specs, fmt.Errorf("Failed to resolve containers: %s", result.JobError.Reason)
```

Comment:
This will lead to a double "Failed to resolve containers":

`ERROR: ContainerResolveError: Failed to resolve containers: failed to resolve container: 'localhost:8080/osbuild/demo@sha256:23b808ef648cee4db38ca39026f1a077507898916e741ceab39563c2dc033bc8': error getting manifest: reading manifest sha256:23b808ef648cee4db38ca39026f1a077507898916e741ceab39563c2dc033bc8 in localhost:8080/osbuild/demo: unauthorized: access to the requested resource is not authorized`